w10h01/test/pgdp/trials/TrialOfTheDreamsTest.java

```diff
@@ -0,0 +1,51 @@
+package pgdp.trials;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import java.util.function.Function;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TrialOfTheDreamsTest {
+    private Function<byte[], Boolean> getLock(byte[] combination) {
+        return (byte[] bytes) -> {
+            if (bytes.length != combination.length) {return false;}
+            for (int i = 0; i < bytes.length; i++) {
+                if (bytes[i] != combination[i]) {return false;}
+            }
+            return true;
+        };
+    }
+
+    @Test
+    public void combinationsTest() {
```

Comment:
```suggestion
    @MethodSource
    @ParameterizedTest
    public void combinationsTest(byte[] combination) {
```

Reply:
We declare it with `@ParameterizedTest` Annotation. This allows us to call tests through a function. `@MethodSource` declares that input will be a static function of same name but empty signature.